packages/beacon-node/src/util/clock.ts

```diff
@@ -97,51 +105,63 @@ export class Clock extends EventEmitter implements IClock {
   get currentSlotWithGossipDisparity(): Slot {
     const currentSlot = this.currentSlot;
     const nextSlotTime = computeTimeAtSlot(this.config, currentSlot + 1, this.genesisTime) * 1000;
-    return nextSlotTime - Date.now() < MAXIMUM_GOSSIP_CLOCK_DISPARITY ? currentSlot + 1 : currentSlot;
+    return nextSlotTime - Date.now() < MAXIMUM_GOSSIP_CLOCK_DISPARITY_MS ? currentSlot + 1 : currentSlot;
   }
 
   get currentEpoch(): Epoch {
     return computeEpochAtSlot(this.currentSlot);
   }
 
-  /** Returns the slot if the internal clock were advanced by `toleranceSec`. */
-  slotWithFutureTolerance(toleranceSec: number): Slot {
-    // this is the same to getting slot at now + toleranceSec
-    return getCurrentSlot(this.config, this.genesisTime - toleranceSec);
+  /** Returns the slot if the internal clock were advanced by `toleranceMs`. */
+  slotWithFutureTolerance(toleranceMs: number): Slot {
```

Comment:
I don't see this function equivalent to old version. Old version returns current slot, or `current slot + 1`. This function return current slot, or `current slot - 1`

Reply:
I added detail doc comment explaining the behavior of the function and the all the usage is also aligned with it.

Reply:
now the doc says `Return the current slot with future tolerance e.g. currentSlot or currentSlot + 1`

but implementation returns `currentSlot or currentSlot - 1`

```typescript
return toleranceMs > 0 && now > currentSlotStartTime && now - currentSlotStartTime <= toleranceMs
      ? this.currentSlot - 1
      : this.currentSlot;
```

Reply:
Please suggest how you feel to describe it well. 

```
* -----------------------------------------------------------------------------------
* |      6          |    7                       |           8                 |
* |                 |<-futureToleranceMs->       |
```

The slot time within the range of `<futureToleranceMs>` will be considered current. If the clock passed `6` and now in the `7` slot but within the range of the `futureToleranceMs` then it will return `6`, not the 7.